### PR TITLE
Add seperate validation for blueprint names

### PR DESCRIPTION
**Pull request by jkozol**
The VALID_API_STRING function allows for characters that should not be
allowed in blueprint names. VALID_BLUEPRINT_NAME allows us to
specifically check if a blueprint contains a valid name.

### CODE REVIEW

**Comment by jkozol** [comment]
@atodorov I am going to add a test to check-api to check that a blueprint with an invalid name returns the proper error.

**Comment by larskarlitski** [comment]
> @atodorov I am going to add a test to check-api to check that a blueprint with an invalid name returns the proper error.

Thanks! Please add a check specifically for the bug that this fixes. I think a `,` in the blueprint name confuses some routes that take multiple blueprints separated by comma? Also mention it in the commit message, to explain why this change is necessary.

File: src/pylorax/api/v0.py [hunk 1]
```diff
@@ -401,8 +401,8 @@ def v0_blueprints_new():
             blueprint = recipe_from_toml(request.data)
         else:
             blueprint = recipe_from_dict(request.get_json(cache=False))
-
-        if VALID_API_STRING.match(blueprint["name"]) is None:
+
```
Comment (bcl) [inline src/pylorax/api/v0.py]: looks like this added some extra whitespace.

**Comment by larskarlitski** [comment]
Travis failure seems unrelated and should be fixed with #797. Please rebase to make sure.

File: src/pylorax/api/v0.py (hunk 1 above)
Comment (jkozol) [reply src/pylorax/api/v0.py]: Thanks for pointing this out. The whitespace should be removed now.

**Pull request merged**